flink-runtime/src/main/java/org/apache/flink/runtime/state/KeyGroupRangeAssignment.java

```diff
@@ -45,7 +47,8 @@ private KeyGroupRangeAssignment() {
 	 * @param parallelism the current parallelism of the operator
 	 * @return the index of the parallel operator to which the given key should be routed.
 	 */
-	public static int assignKeyToParallelOperator(Object key, int maxParallelism, int parallelism) {
+	public static int assignKeyToParallelOperator(@Nullable Object key, int maxParallelism, int parallelism) {
```

Comment:
We should only annotate fields/parameters with `@Nullable` if we allow them to be `null`. This is here not the case.